### PR TITLE
Add support for opencv image manipulation

### PR DESCRIPTION
**Pull request by etiennedm**
If the opencv binding is available (cv2 package), use opencv to perform the concatenation. This consistently results in 1/2 to 1/4 the processing time versus PIL on the few tests I've done.

Things to think through:
1. Add pip3 install opencv-python, and sudo apt install libqtgui4 libqt4-test in documentation
2. OR let the user compile opencv on their raspberry (takes forever)
3. Add tests to spot regressions between PIL and opencv generated images

### CODE REVIEW

File: pibooth/pictures/concatenate.py [hunk 1]
```diff
@@ -41,170 +100,185 @@ def concatenate_pictures_portrait(pictures, footer_texts, bg_color, text_color,
...
+        footer_size = 600 if portrait else 300
+        matrix_width, matrix_height = final_width, final_height - footer_size
+
+    return final_width, final_height, matrix_width, matrix_height, footer_size
+
+
+def concatenate_pictures_PIL(portrait, pictures, footer_texts, bg_color, text_color, footer_fonts, inter_width=None):
+    """ Merge up to 4 PIL images.
+    """
+    start = time.time()
```
Comment (anxuae) [inline pibooth/pictures/concatenate.py]: use pibooth.utils.timeit() function

File: pibooth/pictures/concatenate.py [hunk 2]
```diff
@@ -41,170 +100,185 @@ def concatenate_pictures_portrait(pictures, footer_texts, bg_color, text_color,
...
-      |     +-+     |     |   +-+  +-+  |     | +-+ +-+ +-+ |     |   +-+ +-+   |
-      |             |     |             |     |             |     |   |3| |4|   |
-      +-------------+     +-------------+     +-------------+     +---+-+-+-+---+
-    """
-    widths, heights = zip(*(i.size for i in pictures))
-
-    # starting here we consider that all the images have the same height and widths
-    if inter_width is None:
-        inter_width = max(heights) // 20
+    start = time.time(); overall_start = start; LOGGER.debug("Creating pictures matrix image with size {}x{}".format(matrix_raw_width, matrix_raw_height))
```
Comment (anxuae) [inline pibooth/pictures/concatenate.py]: use pibooth.utils.timeit() function (several can be nested)

File: pibooth/pictures/concatenate.py (hunk 1 above)
Comment (etiennedm) [reply pibooth/pictures/concatenate.py]: Done

File: pibooth/pictures/concatenate.py (hunk 2 above)
Comment (etiennedm) [reply pibooth/pictures/concatenate.py]: Done

**Pull request merged**